### DISCUSSION TITLE
WalletConnectError: Unsupported methods (Flutter)

### DISCUSSION THREAD

**Discussion by kinga112** [WalletConnect/walletconnect-monorepo#3418] [Developer Support] [answered]
I was able to create and connect to sign client, but when I requested a signature using signClient.request with eth_signTransaction as the method, I get an error. I tried personal_sign as well.
Unhandled Exception: WalletConnectError(code: 5101, message: Unsupported methods. The method eth_signTransaction is not supported, data: null)

**Comment by glitch-txs** [comment]
Hi! `eth_signTransaction` is not supported by most wallets, personal_sign should be fine, which wallet did you tested with?

**Reply by kinga112** [reply]
I am using metamask. I tried personal_sign and I get the same error: Unhandled Exception: WalletConnectError(code: 5101, message: Unsupported methods. The method personal_sign is not supported, data: null)

**Comment by glitch-txs** [comment]
Could you try on labs?
- https://lab.web3modal.com/with-wagmi/react

Also we have this demo, both websites have personal_sign
- https://react-app.walletconnect.com/

Just tested MetaMask on android and works fine, is it android or iOS?

**Reply by kinga112** [reply]
I guess since my signClient connect used ['eth_sendTransation', 'personal_sign'] it still failed. I removed eth_sendTransation and personal sign is not givving an issue anymore.  Pairing _onMessageEvent, Received data: {id: 1691627443850898, jsonrpc: 2.0, error: {code: 5000, message: {"code":-32602,"message":"Invalid parameters: must provide an Ethereum address."}}} Im getting this error now. I am not sure where I am suppose to input address. Here is my code: 

final signature = await signClient.request(
    topic: session.topic,
    chainId: 'eip155:1',
    request: const SessionRequestParams(
      method: 'personal_sign',
      params: 'json serializable parameters',
    ),
  );

**Reply by kinga112** [reply, marked as answer]
I was able to add the address and change to params to: params: ['test', address]),
